File: models/subject.go
package models

type Subject struct {
	Base
	Name      string `json:"name" gorm:"type:varchar(32) not null"`
	ShortName string `json:"short_name" gorm:"type:varchar(16) not null"`
	Url       string `json:"url"`
	GroupId   int32  `json:"-"`
	Group     Group  `json:"group"`
	Rooms     []Room `gorm:"many2many:subjects_rooms;" json:"rooms"`
}

File: src/models/Subject.ts
import Group from "./Group";
import Common from "./Common";
import { Model } from "objection";
import Room from "./Room";

export default class Subject extends Common {
  name!: string;
  short_name!: string;
  url!: string;
  group!: Group;
  rooms: Room[] = [];

  static get tableName() {
    return 'subjects';
  }

  static get relationMappings() {
    return {
      group: {
        relation: Model.HasOneRelation,
        modelClass: Group,
        join: {
          from: 'groups.id',
          to: 'subjects.group_id'
        }
      },
      rooms: {
        relation: Model.ManyToManyRelation,
        modelClass: Room,
        join: {
          from: 'subjects.id',
          through: {
            from: 'subjects_rooms.subject_id',
            to: 'subjects_rooms.room_id'
          },
          to: 'rooms.id'
        }
      }
    };
  }
}
